packages/core/host/networking/autohotspot/overlays/usr/libexec/autohotspot

```diff
@@ -12,29 +12,27 @@
 
 wifi_dev="wlan0" # device name to use; by default wlan0 is the on-board wifi chip on the RPi4
 
-# Determine the SSIDs of the wifi networks the RPi is configured to use:
-# FIXME: can we simplify these commands?
-wpa_ssid=$(grep -e '^[[:space:]]*ssid="' /etc/wpa_supplicant/wpa_supplicant.conf | \
-  awk -F'ssid=' '{ print $2 }' | sed 's/\r//g' | \
-  awk 'BEGIN{ORS=","} {print}' | sed 's/\"//g' | sed 's/,$//')
-IFS="," read -r -a ssids <<<"$wpa_ssid"
+hostname="$(sed 's~#.*$~~g' /etc/hostname | tr -d '[:space:]')"
+hostname_32="${hostname:0:32}"
+# list of ssids to connect to
+ssids=$(nmcli -t -f NAME,TYPE connection show | grep "802-11-wireless" | grep -v "$hostname_32" | cut -d':' -f1 | sort | uniq)
```

Comment:
As discussed in our meeting last week, we need to sort this list by connection timestamp instead, so that (once we add proper RTC support) the most-recently-used network will be preferred.